components/shapediver/modelState/ModelStateNotificationCreated.tsx

```diff
@@ -22,11 +22,18 @@ export default function ModelStateNotificationCreated({
 }: IModelStateNotificationCreatedProps) {
 	const notifications = useContext(NotificationContext);
 
-	const copyStateLink = useCallback(() => {
-		navigator.clipboard.writeText(link);
-		notifications.success({
-			message: "State link copied to clipboard!",
-		});
+	const copyStateLink = useCallback(async () => {
+		try {
+			await navigator.clipboard.writeText(link);
+			notifications.success({
+				message: "State link copied to clipboard!",
+			});
+		} catch {
+			/* clipboard write failed */
+			notifications.warning({
+				message: `State link could not be copied to clipboard, please copy it from here: ${link}`,
+			});
+		}
```

Comment:
@mathieu-shapediver FYI inside the Shopify iframe, copying to the clipboard seems to be forbidden due to some security policy.